builtins/web/blob.cpp

```diff
@@ -474,35 +474,40 @@ bool Blob::stream_pull(JSContext *cx, JS::CallArgs args, JS::HandleObject source
   return true;
 }
 
-std::vector<uint8_t> *Blob::blob(HandleObject self) {
+std::vector<uint8_t> *Blob::blob(JSObject *self) {
+  MOZ_ASSERT(is_instance(self));
   auto blob = static_cast<std::vector<uint8_t> *>(
       JS::GetReservedSlot(self, static_cast<size_t>(Blob::Slots::Data)).toPrivate());
 
   MOZ_ASSERT(blob);
   return blob;
 }
 
-size_t Blob::blob_size(HandleObject self) {
+size_t Blob::blob_size(JSObject *self) {
+  MOZ_ASSERT(is_instance(self));
   auto blob = static_cast<std::vector<uint8_t> *>(
       JS::GetReservedSlot(self, static_cast<size_t>(Blob::Slots::Data)).toPrivate());
 
   MOZ_ASSERT(blob);
   return blob->size();
```

Comment:
I just realized that you can reduce this function to just be this:
```suggestion
  return blob(self)->size();
```

Reply:
Nice!